src/Context/Settings/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, {useState, useEffect} from 'react';
+
+
+// create a context object.......
+
+export const SettingsContext = React.createContext();
+
+// create a provider component
+function SettingsProvider({ children }){
+    const [ displayCount, setDisplayCount ] = useState(3);
+    const [ showCompleted, setShowCompleted ] = useState(false);
+    const [ sortField, setSortField ] = useState('difficulty');
+
+  // Read settings from Local Storage on component mount
+  useEffect(() => {
+    const storedSettings = localStorage.getItem('settings');
+    if (storedSettings) {
+      const { displayCount, showCompleted, sortField } = JSON.parse(storedSettings);
+      setDisplayCount(displayCount);
+      setShowCompleted(showCompleted);
+      setSortField(sortField);
+    }
+  }, []);
+
+    // Update Local Storage when settings change
+    useEffect(() => {
+        const settings = {
+          displayCount,
+          showCompleted,
+          sortField,
+        };
+        localStorage.setItem('settings', JSON.stringify(settings));
+      }, [displayCount, showCompleted, sortField]);
```

Comment:
I'm not totally sure what's going on with ordering and execution time (it might have something to do with the React dev environment unmount/remount) but it seems like this is setting local storage with the initialized values...